### DISCUSSION TITLE
How can i get the C/C++ source code of the swig files(such as fitz.i ) ?

### DISCUSSION THREAD

**Discussion by JoyChenNo1** [pymupdf/PyMuPDF#2354] [Q&A] [answered]

**Comment by JorjMcKie** [comment]
Look here: https://github.com/pymupdf/PyMuPDF/tree/master/fitz

**Comment by JoyChenNo1** [comment]
> Look here: https://github.com/pymupdf/PyMuPDF/tree/master/fitz

I want to directly use some functions like xref_get_keys in my C++ source code(use mupdf), but I just found the swig file in the fitz.i.How can I get those C implement functions?

**Comment by JorjMcKie** [comment]
This is a genuine function of PyMuPDF (not MuPDF). You will find it in `fitz.i`.

**Reply by JorjMcKie** [reply, marked as answer]
It is written in C. You may also look at `fitz_wrap.c` of the source distribution.

**Reply by JoyChenNo1** [reply]
Thanks a lot, I get this c file by `swig`
